src/HasMedia/HasMediaTrait.php

```diff
@@ -411,7 +413,9 @@ protected function mediaIsPreloaded(): bool
      */
     public function loadMedia(string $collectionName)
     {
-        return $this->media
+        $collection = $this->exists ? $this->media : collect($this->unsavedMediaItems);
```

Comment:
Could you add a testcase that proves this works when the model hasn't been saved yet?